pennylane/gradients/parameter_shift.py

```diff
@@ -756,6 +755,16 @@ def var_param_shift(tape, argnum, shifts=None, gradient_recipes=None, f0=None, b
     return gradient_tapes, processing_fn
 
 
+def _param_shift_stopping_condition(op) -> bool:
+    if not op.has_decomposition:
+        # let things without decompositions through without error
+        # error will happen when calculating parameter shift tapes
+        return True
+    if isinstance(op, qml.operation.Operator) and any(qml.math.requires_grad(p) for p in op.data):
+        return op.grad_method is not None

```

Comment:
I'm wondering whether we should change the behaviour here, at some point: `param_shift` has a fallback to `finite_diff` for operations that have `grad_method="F"`, but maybe it would make sense to decompose such operations, with the hope to get `grad_method="A"` :thinking: Definitely out of scope!